npm/vite-dev-server/src/makeCypressPlugin.ts

```diff
@@ -51,39 +48,53 @@ export const makeCypressPlugin = (
   const posixSupportFilePath = supportFilePath ? convertPathToPosix(resolve(projectRoot, supportFilePath)) : undefined
   const posixIndexHtml = indexHtml ? convertPathToPosix(resolve(projectRoot, indexHtml)) : undefined
 
-  const normalizedSupportFilePath = posixSupportFilePath ? `${base}@fs/${posixSupportFilePath}` : undefined
-
   return {
     name: pluginName,
     enforce: 'pre',
-    config (_, env) {
-      if (env) {
-        return {
-          define: {
-            'import.meta.env.__cypress_supportPath': JSON.stringify(normalizedSupportFilePath),
-            'import.meta.env.__cypress_originAutUrl': JSON.stringify(`${namespace}/iframes/${convertPathToPosix(projectRoot)}/`),
-          },
-        }
-      }
-    },
     configResolved (config) {
       base = config.base
     },
     async transformIndexHtml () {
       debug('transformIndexHtml with base', base)
       const indexHtmlPath = indexHtml ? resolve(projectRoot, indexHtml) : resolve(__dirname, '..', 'index.html')
-      const indexHtmlContent = await read(indexHtmlPath, { encoding: 'utf8' })
-
-      return {
-        html: indexHtmlContent,
-        // load the script at the end of the body
-        // script has to be loaded when the vite client is connected
-        tags: [{
-          tag: 'script',
-          injectTo: 'body',
-          attrs: { type: 'module' },
-          children: `import(${JSON.stringify(`${base}@fs/${INIT_FILEPATH}`)})`,
-        }],
+      const indexHtmlContent = await readFile(indexHtmlPath, { encoding: 'utf8' })
+      // find </body> last index
+      const endOfBody = indexHtmlContent.lastIndexOf('</body>')
+
+      // insert the script in the end of the body
+      return `${indexHtmlContent.substring(0, endOfBody)
+    }<script src="${base}cypress:client-init-test" type="module"></script>${
+      indexHtmlContent.substring(endOfBody)
+    }`
+    },
+    resolveId (id) {
+      if (id === 'cypress:config') {
+        return id
+      }
+
+      if (id === 'cypress:support-path') {
+        return posixSupportFilePath
+      }
+
+      if (id === 'cypress:spec-loaders') {
+        return id
+      }
+
+      if (id === '/cypress:client-init-test') {
+        return INIT_FILEPATH
+      }
+    },
+    load (id) {
+      if (id === 'cypress:spec-loaders') {
+        return `export default {\n${specs.map((s) => {
+          return `${JSON.stringify(s.relative)}:()=>import(${JSON.stringify(s.absolute)})`
+        }).join(',\n')}\n}`
+      }
+
+      if (id === 'cypress:config') {
+        return `
+export const hasSupportPath = ${JSON.stringify(!!supportFilePath)}
+export const originAutUrl = ${JSON.stringify(`/${namespace}/iframes/${normalizePath(projectRoot)}/`)}`
       }
     },
     configureServer: async (server: ViteDevServer) => {
```

Comment:
@elevatebart I didn't pull forward your configureServer changes; @ZachJW34 had also made changes in 10.0 and they seemed to be equivalent. Let me know if that is not the case.